api/v1beta1/hcloudmachinetemplate_webhook.go

```diff
@@ -67,7 +67,7 @@ func (r *HCloudMachineTemplateWebhook) ValidateUpdate(ctx context.Context, oldRa
 		return nil, apierrors.NewBadRequest(fmt.Sprintf("expected a admission.Request inside context: %v", err))
 	}
 
-	var allErrs field.ErrorList
+	allErrs := validateHCloudMachineSpec(oldHCloudMachineTemplate.Spec.Template.Spec, newHCloudMachineTemplate.Spec.Template.Spec)
```

Comment:
this validation should go below the code block that comes after this one. 

If we have not this "should skip immutability check", then we don't allow any update anyway. 

Same goes for the hetznerbaremetalmachine